src/services/chains.rs

```diff
@@ -0,0 +1,36 @@
+use crate::cache::cache_operations::RequestCached;
+use crate::config::{
+    base_config_service_url, chain_info_cache_duration, chain_info_request_timeout,
+};
+use crate::models::chains::ChainInfo;
+use crate::models::commons::Page;
+use crate::utils::context::Context;
+use crate::utils::errors::ApiResult;
+
+pub async fn get_chains_paginated(
+    context: &Context<'_>,
+    limit: Option<&String>,
+    offset: Option<&String>,
+) -> ApiResult<Page<ChainInfo>> {
+    let mut queries: Vec<(String, String)> = vec![];
+    if let Some(limit) = limit {
+        queries.push(("limit".to_string(), limit.to_owned()))
```

Comment:
I think we always use `to_string` instead of `to_owned`. We should stick to this (at least in this method :P )

Reply:
I would also use `HashMap` here and maybe `<&str, String>` instead, so you don't have have to call `to_string` on the key too.

Reply:
Changed to `HashMap` 👍 bde0d85